dts/bindings/pinctrl/st,stm32-pinctrl.yaml

```diff
@@ -54,41 +54,10 @@ child-binding:
       description: |
         Reused from
         https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml
-        Integer array, represents gpio pin number and mux setting.
-        These defines are calculated as: ((port * 16 + line) << 8) | function
-        With:
-        - port: The gpio port index (PA = 0, PB = 1, ..., PK = 11)
-        - line: The line offset within the port (PA0 = 0, PA1 = 1, ..., PA15 = 15)
-        - function: The function number, can be:
-        * 0 : Alternate Function 0
-        * 1 : Alternate Function 1
-        * 2 : Alternate Function 2
-        * ...
-        * 15 : Alternate Function 15
-        * 16 : Analog
-        * 17 : GPIO
-        In case selected pin function is GPIO, pin is statically configured as
-        a plain input/output GPIO. Default configuration is input. Output value
-        can be configured by adding 'ouptut-low' or 'output-high' properties
-        to the pin configuration.
-
-        To simplify the usage, macro is available to generate "pinmux" field.
-        This macro is available here:
-          -include/zephyr/dt-bindings/pinctrl/stm32-pinctrl-common.h
-        Some examples of macro usage:
-           GPIO A9 set as alternate function 2
-        ... {
-                 pinmux = <STM32_PINMUX('A', 9, AF2)>;
-        };
-           GPIO A9 set as analog
-        ... {
-                 pinmux = <STM32_PINMUX('A', 9, ANALOG)>;
-        };
-           GPIO A9 set as GPIO output high
-        ... {
-                 pinmux = <STM32_PINMUX('A', 9, GPIO)>;
-                 output-high;
-        };
+        Integer represents gpio pin number and mux setting.
+        Each integer encodes a port, line and alternate function.
+        Refer to STM32_PINMUX() for details about encoding in
+        include/zephyr/dt-bindings/pinctrl/stm32-pinctrl.h
```

Comment:
Well, now that's a bit too much removal of information.
To me we can only remove `These defines are calculated as: ((port * 16 + line) << 8) | function` and the rest could stay.
We might just need to update the header path (currently provided as "include/zephyr/dt-bindings/pinctrl/stm32-pinctrl-common.h").